### ISSUE TITLE
Ping/Pong packets not being translated properly to Transactions (1.17 server/< 1.17 client)

### ISSUE THREAD

**Issue by freppp** [ViaVersion/ViaVersion#2624]
### `/viaversion dump` Output

https://dump.viaversion.com/5c74a19131546496df1372f29f51c60b56235a3079b20cffc1d8fe2c83d42a17

### Console Error

```
Put the paste.gg link or text here.
```

### Bug Description

Ping/Pong packets not being properly translated to transactions on a 1.17 server and a client older than 1.17.

### Steps to Reproduce

Login with a < 1.17 client on a 1.17 server with ViaVersion/ViaBackwards/ViaRewind

### Expected Behavior

1.16.5: https://cdn.discordapp.com/attachments/854498117215060048/864358462649139231/unknown.png
1.17: https://i.imgur.com/qWMppIt.png

my server is on 1.17.1 spigot, top ss is from a 1.16.5 client and bottom ss is from a 1.17 client

on 1.16.5 and older servers i use transaction packets to verify a bunch of stuff and to accurately determine client ping and in 1.17 and newer i use ping/pong packets, and i know via translates them to be accurate depending on the users client version. for some reason on 1.16.5 it seems something is wrong (it could be on my end but ive been searching for a while and have come up short) that the pings that are sent receive a pong back instantly within the same millisecond/tick no matter what (ive used clumsy to simulate lagging and it is still always 0-1ms delay) but as soon as i log onto 1.17 everything works normally and my ping is calculated at 30-40ms

### Checklist

- [X] Via plugins are only running on **EITHER** the backend servers (e.g. Paper) **OR** the proxy (e.g. BungeeCord), **not on both**.
- [X] I have included a ViaVersion dump.
- [X] If applicable, I have included a paste (**not a screenshot**) of the error.
- [X] I have tried a build from https://ci.viaversion.com/ and the issue still persists.

**Comment by Barvalg** [comment]
![Platform: git-Paper-98 (MC: 1.17.1)](https://img.shields.io/badge/Platform-git--Paper--98%20%28MC%3A%201.17.1%29-3498db?style=flat-square) <br />![ViaVersion (4.0.1): 2 commits behind master](https://img.shields.io/badge/ViaVersion%20(4.0.1)-2%20commit(s)%20behind%20master-yellow?style=flat-square)<br />![ViaBackwards(4.0.1): 4 commits behind master](https://img.shields.io/badge/ViaBackwards%20(4.0.1)-4%20commit(s)%20behind%20master-yellow?style=flat-square)<br />![ViaRewind(2.0.1): 2 commits behind master](https://img.shields.io/badge/ViaRewind%20(2.0.1)-2%20commit(s)%20behind%20master-yellow?style=flat-square)

**Comment by Camotoy** [comment]
This belongs in the ViaBackwards repository, but I believe the consensus that the team reached was that this can't reasonably happen: https://github.com/ViaVersion/ViaBackwards/pull/376

TL;DR: the problem is that the ping/pong packets use an integer, but the inventory confirmation packet uses a short, which can't store as large of a number. Therefore VB can't reliably send over every single packet (sure, it can forward ones that fit, but does it cancel any integer that's too large?). I don't believe ViaBackwards has an API option for listening to inventory confirmation packets, but [outsider opinion] I think this isn't an unreasonable request to add one for anticheats to use.

**Comment by MWHunter** [comment]
I would be perfectly fine with only being able to send shorts but not integers.  I can account for that and already only send shorts even with the PONG packet for consistency.  I can't account for the packet entirely missing as that causes the velocity check to fail 100% of the time over 100 ms ping and other major latency issues.

**Comment by freppp** [comment]
> This belongs in the ViaBackwards repository, but I believe the consensus that the team reached was that this can't reasonably happen: [ViaVersion/ViaBackwards#376](https://github.com/ViaVersion/ViaBackwards/pull/376)
> 
> TL;DR: the problem is that the ping/pong packets use an integer, but the inventory confirmation packet uses a short, which can't store as large of a number. Therefore VB can't reliably send over every single packet (sure, it can forward ones that fit, but does it cancel any integer that's too large?). I don't believe ViaBackwards has an API option for listening to inventory confirmation packets, but [outsider opinion] I think this isn't an unreasonable request to add one for anticheats to use.

I (and pretty much every other anticheat dev) would pretty much only send pings with short values anyways since I just copy/pasted how I do it with transactions and just changed which packet I send lol. If this could be added/fixed It would be a really great feature imo

**Comment by kennytv** [comment]
I'll look into adding a config option/system property to transform ping packets in the short range to transaction packets, mapping those to pong packets when acknowledged. With this, you'd have to do the least work on your side (=only using shorts when sending ping packets to a client < 1.17).

**Comment by freppp** [comment]
> I'll look into adding a config option/system property to transform ping packets in the short range to transaction packets, mapping those to pong packets when acknowledged. With this, you'd have to do the least work on your side (=only using shorts when sending ping packets to a client < 1.17).

That would be amazing. Thanks so much

**Comment by kennytv** [comment]
You can either tell users to enable the option in the ViaBackwards config, or use `System.setProperty("com.viaversion.handlePingsAsInvAcknowledgements", "true");` in your plugin. As said, the one restriction is that you have to use ping ids in the short range.
